streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -70,6 +70,7 @@ public class StreamsProducer {
     private Producer<byte[], byte[]> producer;
     private boolean transactionInFlight = false;
     private boolean transactionInitialized = false;
+    private boolean allowReset = true;
```

Comment:
it's a bit confusing that we flip-flop back and forth between "disableReset" and "allowReset", I'd recommend picking one (probably `disableReset`) for all the method and variable names